environment

```diff
@@ -30,13 +30,10 @@ _select () {
 # https://github.com/boto/boto3/issues/1179#issuecomment-569940530
 #
 _preauth () {
-    if ! gcloud auth application-default print-access-token >/dev/null 2>&1; then
-        gcloud auth application-default login --project="$GOOGLE_PROJECT"
-    else
-        gcloud auth application-default set-quota-project "$GOOGLE_PROJECT"
-    fi
-    echo "Google application default credentials are in effect."
-    echo "Use 'gcloud auth application-default revoke' to remove them."
+    gcloud config set project "$GOOGLE_PROJECT"
+    while ! gcloud auth application-default print-access-token >/dev/null 2>&1; do
+        gcloud auth application-default login
+    done
```

Comment:
Sorry, I changed my mind on this. I'm worried that this will get us into an infinite loop. It would be better to check once for the need to invoke `gcloud auth` and if there is, invoked and afterwards check once more to ensure that the post-condition is true. If the second check fails, _preauth should fail.